<launch>
	<include file="$(find rosbridge_server)/launch/rosbridge_websocket.launch">
		<arg name="port" value="9090"/>
	</include>

	<node name="file_server" pkg="file_server" type="file_server" output="screen"/>

    <arg name="mocap_server" default="192.168.1.202"/>

	<node pkg="vrpn_client_ros" type="vrpn_client_node" name="vrpn_client_node" output="screen">
        <rosparam subst_value="true">
            server: $(arg mocap_server)
            port: 3883

            update_frequency: 100.0
            frame_id: world

            # Use the VRPN server's time, or the client's ROS time.
            use_server_time: false
            broadcast_tf: true

            # Must either specify refresh frequency > 0.0, or a list of trackers to create
            #refresh_tracker_frequency: 1.0
            trackers:
            - rosbot01
            - rosbot02
            - cf2
        </rosparam>
        <remap from="/vrpn_client_node/rosbot01/pose" to="/rosbot01/pose"/>
        <remap from="/vrpn_client_node/rosbot02/pose" to="/rosbot02/pose"/>
        <remap from="/vrpn_client_node/cf2/pose" to="/cf2/pose"/>
    </node>

    <node name="ROSBotSwarm" pkg="adam_thesis" type="ROSBotSwarm.py" output="screen"/>

</launch>
